framework/src/problems/FEProblemBase.C

```diff
@@ -4473,7 +4501,15 @@ FEProblemBase::addTransfer(const std::string & transfer_name,
   if (parameters.get<ExecFlagEnum>("execute_on").contains(EXEC_SAME_AS_MULTIAPP))
   {
     ExecFlagEnum & exec_enum = parameters.set<ExecFlagEnum>("execute_on", true);
-    std::shared_ptr<MultiApp> multiapp = getMultiApp(parameters.get<MultiAppName>("multi_app"));
+    std::shared_ptr<MultiApp> multiapp;
+    if (parameters.isParamValid("multi_app"))
+      multiapp = getMultiApp(parameters.get<MultiAppName>("multi_app"));
+    else if (parameters.isParamValid("from_multi_app"))
+      multiapp = getMultiApp(parameters.get<MultiAppName>("from_multi_app"));
+    else if (parameters.isParamValid("to_multi_app"))
+      multiapp = getMultiApp(parameters.get<MultiAppName>("to_multi_app"));
+    else
+      mooseError("Should not reach here");
```

Comment:
we need to have a better error message here. A user specified `to_multiapp` as their parameter name and got the "Should not reach here" error

Reply:
oh wait ... it's just that these parameter names *here* are actually wrong. The user had specified the right parameter name in `to_multiapp`

Reply:
It might be helpful to output users' input

Reply:
my bad.
How are they hitting this? All the apps were moved to the new system

Reply:
well actually, how are they hitting this just now?
is EXEC_SAME_AS_MULTIAPP just not tested?

Reply:
I misread the situation. See https://github.com/idaholab/moose/pull/20803